merlin-server/src/test/java/gov/nasa/jpl/aerie/merlin/server/services/TypescriptCodeGenerationServiceTest.java

```diff
@@ -1,17 +1,82 @@
 package gov.nasa.jpl.aerie.merlin.server.services;
 
+import gov.nasa.jpl.aerie.merlin.server.mocks.StubMissionModelService;
 import org.junit.jupiter.api.Test;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class TypescriptCodeGenerationServiceTest {
 
   @Test
-  void testCodeGen() {
+  void testCodeGen() throws MissionModelService.NoSuchMissionModelException {
+    final var codeGenService = new TypescriptCodeGenerationService(new StubMissionModelService());
+
     assertEquals(
         """
             /** Start Codegen */
+            import * as AST from './constraints-ast.js';
+            import { Discrete, Real, Windows } from './constraints-edsl-fluent-api.js';
+            export type ActivityTypeName =
+              | "activity"
+              ;
+            export type ResourceName = "mode" | "state of charge";
+            export type DiscreteResourceSchema<R extends ResourceName> =
+              R extends "mode" ? ( | "Option1" | "Option2") :
+              R extends "state of charge" ? number :
+              void;
+            export function discreteResourceSchemaDummyValue<R extends ResourceName>(resource: R): DiscreteResourceSchema<R> {
```

Comment:
I'm confused why this is here. I know its used in the constructor for Discrete, but I'm struggling to understand the purpose there too. It seems unused.

Reply:
You highlighted multiple lines, I assume you are talking about `discreteResourceSchemaDummyValue`? That is so in the `Discrete.Resource` function it can get an instance of the discrete schema to assign to `this.__schemaInstance`. The `DummyValue` function has to be generated code because the dummy value's type depends on the mission model's resource types.

It is only for type checking, and the value is technically unused. But if it isn't stored in `this.__schemaInstance`, then `Discrete<number>` can be assigned to `Discrete<string>` or any other `Discrete` type erroneously.

Reply:
IIRC this one failed when I removed the field:
```typescript
export default () => {
  return Constraint.ForEachActivity(
    "activity",
    (alias) => Discrete.Resource("mode").equal(alias.parameters().Param)
  )
}
```

This is using the `StubMissionModel` mock class, where `mode` is either `"Option1"` or `"Option2"`, and the `Param` parameter of `activity` is any `string`.

Reply:
That's the very last test case in `ConstraintsDSLCompilationServiceTests`. If you remove the field it will fail.

Reply:
I just removed the dummy value, and all tests pass.

Reply:
You might have not re-run the tests because they were cached. I just tried again. If you remove the field it fails, even if you keep the extra constructor argument.

Reply:
resolving, see [this comment](https://github.com/NASA-AMMOS/aerie/pull/183#discussion_r882873669)